huaweicloud/services/acceptance/workspace/data_source_huaweicloud_workspace_tags_test.go

```diff
@@ -0,0 +1,175 @@
+package workspace
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccWorkspaceTagsDataSource_basic(t *testing.T) {
+	var (
+		rName = acceptance.RandomAccResourceNameWithDash()
+
+		desktop      interface{}
+		resourceName = "huaweicloud_workspace_desktop.test"
+		rc           = acceptance.InitResourceCheck(
+			resourceName,
+			&desktop,
+			getDesktopFunc,
+		)
+
+		dcName = "data.huaweicloud_workspace_tags.all"
+		dc     = acceptance.InitDataSourceCheck(dcName)
+
+		dcByKeyName = "data.huaweicloud_workspace_tags.filter_by_key"
+		dcByKey     = acceptance.InitDataSourceCheck(dcByKeyName)
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWorkspaceTagsDataSource_base(rName),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(dcName, "tags.#"),
+					resource.TestCheckResourceAttrSet(dcName, "tags.0.key"),
+					resource.TestCheckResourceAttrSet(dcName, "tags.0.values.#"),
+					dcByKey.CheckResourceExists(),
+					resource.TestCheckOutput("is_key_filter_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccWorkspaceTagsDataSource_base(rName string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_workspace_tags" "all" {
+  depends_on = [
+    huaweicloud_workspace_desktop.test
+  ]
+}
+
+locals {
+  key = "terraform_greyworld"
+}
+
+data "huaweicloud_workspace_tags" "filter_by_key" {
+  depends_on = [
+    huaweicloud_workspace_desktop.test
+  ]
+
+  key = "terraform_greyworld"
+}
+
+locals {
+  tag_filter_result = [
+     for v in data.huaweicloud_workspace_tags.filter_by_key.tags[*].key : v == local.key
+  ]
+}
+
+output "is_key_filter_useful" {
+  value = length(local.tag_filter_result) > 0 && alltrue(local.tag_filter_result)
+}
+`, applyAccDesktop(rName))
+}
+
+func applyAccDesktop_base() string {
```

Comment:
```suggestion
func applyAccDesktop_base() string {
```
testAccTagsDataSource_base